7 Sprint

AlexSunset opened the pull request.

Commit:
7 Sprint

Commit:
Allure Report

alesyaisachenko commented on src/test/java/CourierCreateTest.java:
```diff
@@ -0,0 +1,71 @@
+    //При создании нового курьера 201 код ответа и "ок": true
+    @Test
+    public void createNewCourierAndCheckResponse() {
+        Response response =
```
Это нужно обязательно исправить: 

нужно улучшить архитектуру кода.

Сейчас в каждом тесте ты дублируешь код по формированию и отправления запроса.

Дублирование кода нужно избегать. Как это сделать:
Создать классы CourierApi и OrderApi.
В каждом классе создать методы для формирования запроса (например: создание courier, удаление courier, логин и т.д.).

В этом случае, если изменится запрос для создания курьера, тебе придется вносить правки в одном месте (в методе), а не во все тесты сразу.

alesyaisachenko commented on src/test/java/CourierFailCreateTest.java:
```diff
@@ -0,0 +1,62 @@
+    CreateCourier createCourier = new CreateCourier(login, password, firstName);
+
+    @Before
+    public void setUp() {
```
Это нужно обязательно пофиксить во всем коде.

этот метод у тебя повторяется в каждом тестовом классе -- это дублирование кода. Дублирование кода нужно избегать.

Как это можно сделать, есть два варианта, выбирай любой:
1. Создать класс BaseTest, в него вынести @before, все остальные тестовые классы наследовать от BaseTest.
2. Вообще в этом методе идет настройка URI, это относится к формированию запроса,  соответственно лучше это вынести в BaseApi, в нем задать RequestSpecification. Классы CourierApi и OrderApi наследовать от BaseApi.

alesyaisachenko commented on src/test/java/CourierLoginTest.java:
```diff
@@ -0,0 +1,119 @@
+                .when()
+                .post("/api/v1/courier/login")
+                .then()
+                .statusCode(200);
```
Можно лучше:

статус коды можно было взять из библиотеки:
import  static  org.apache.http.HttpStatus.*;

statusCode(SC_CREATED)

alesyaisachenko commented on src/test/java/CourierCreateTest.java:
```diff
@@ -0,0 +1,71 @@
+                        .and()
+                        .body(createCourier)
+                        .when()
+                        .post("/api/v1/courier");
```
Можно лучше :

"/api/v1/courier" -- лучше было вынести в final static переменную.

alesyaisachenko commented on src/test/java/CreateOrderTest.java:
```diff
@@ -0,0 +1,95 @@
+    @Test
+    public void greyColourOrderCreate(){
+        ArrayList<String> testColour = new ArrayList<>();
+        testColour.add("GREY");
```
Это нужно доработать:

эти тесты лучше реализовать через параметризацию.

alesyaisachenko commented on src/test/java/CourierCreateTest.java:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    //При создании нового курьера 201 код ответа и "ок": true
+    @Test
```
Это нужно обязательно доработать:

Нужно настроить алюр-отчет в коде.

Commit:
Испрвления по ревью:
- Вынес в BaseTest общие переменные и инициализацию
- Для обращения к эндпоинтам создал классы CourierApi и OrderApi, в которых обращения реализованы в виде методов
- Сделал параметризованный тест для создания заказа с разными цветами самокатов
- Настроил описание методов в отчёте Allure
- Заменил числовое обозначение статус кодов ответов на соответсвующие константы из interface HttpStatus

alesyaisachenko commented on src/main/java/CourierApi.java:
```diff
@@ -0,0 +1,27 @@
+
+    public final static String CREATE_COURIER_ENDPOINT = "/api/v1/courier";
+    public final static String LOGIN_COURIER_ENDPOINT = "/api/v1/courier/login";
+
```
Можно лучше:

используй аннотацию [@Step](https://github.com/step), для того чтобы отчеты в алюр были более информативные.

alesyaisachenko commented on src/main/java/CreateCourierJson.java:
```diff
@@ -0,0 +1,40 @@
+    private String password;
+    private String firstName;
+
+    public CreateCourierJson(String login, String password, String firstName) {
```
Можно лучше:

CreateCourierJson -- не удачное название класса, в результате не получится json, получится POJO

alesyaisachenko commented on src/test/java/CourierCreateTest.java:
```diff
@@ -0,0 +1,43 @@
+
+    @After
+    public void deleteCourier(){
+        int id = given()
```
Это нужно доработать:

весь код для формирования запросов должен лежать в специальном классе CourierApi

alesyaisachenko commented on src/test/java/OrdersListTest.java:
```diff
@@ -0,0 +1,30 @@
+public class OrdersListTest extends BaseTest {
+
+    @Test
+    public void getOrdersReturnOrders(){
```
Это нужно доработать:

весь код для формирования запросов должен лежать в специальном классе OrderApi

Commit:
- Доделал по критичным замечаниям
- Учёл про пакеты на будущее и про аннтоцию Step
- Поменял название классов с Json на POJO
